crypto/asn1/asn_moid.c

```diff
@@ -99,7 +99,13 @@ static int do_create(const char *value, const char *name)
         lntmp[p - ln] = 0;
         oid = OBJ_nid2obj(nid);
         oid->ln = lntmp;
```

Comment:
If there is any leak here, it is due to this line of code.
The original value is lost.
Your change just discards the job of `if (p) {` statement.
And so it loses both original and modified long names (ln) .